src/scenes/app/stores/index.js

```diff
@@ -1,38 +1,65 @@
-import { observable, action } from 'mobx';
-import sampleData from '../../../helpers/dataSourceSample';
+import { observable, action, toJS } from 'mobx';
 import strings from '../../../localization/en/strings';
+import Utils from '../../../helpers/Utils';
 
 class ObservableTodoStore {
-  @observable items = sampleData;
+  @observable items = [];
   @observable error = '';
+  @observable start = true;
 
   @action
   clearAllDone = () => {
-    this.items = this.items.filter(element => element.done === false);
+    this.items = this.items.filter(element => element.completed === false);
+  }
+
+  getToDosDone = () => {
+    return this.items.filter(element => element.completed === true);
+  }
+
+  getToDoById = (id) => {
+    return toJS(this.items.find(element => element.id === id));
   }
 
   @action
-  addToDo = (newTitle, newDescription) => {
+  addToDo = (newToDo) => {
+    const id = { id: Utils.getId(newToDo.url) };
+    const toDoWithId = { ...newToDo, ...id };
+    this.items.push(toDoWithId);
+  }
+
+  @action
+  validateToDo = (title) => {
     const { emptyTextOrDescription } = strings;
-    if (newTitle.length !== 0 && newDescription.length !== 0) {
-      const newToDo = {
-        id: `${newTitle}${newDescription}`,
-        title: newTitle,
-        description: newDescription,
-        done: false,
-      };
-      this.items.push(newToDo);
-      this.error = '';
-    } else {
+    this.error = '';
+    if (title.length === 0) {
       this.error = emptyTextOrDescription;
     }
-    return newTitle.length !== 0 && newDescription.length !== 0;
+    return title.length !== 0;
+  }
+
+  @action
+  addItems = (newItems) => {
+    newItems.forEach((toDo) => {
+      this.addToDo(toDo);
+    });
```

Comment:
Maybe this could be written like `newItems.forEach(toDo => this.addTodo(toDo))` and we can remove the unnecessary braces. What do you think?